exercises/concept/lasagna/.docs/introduction.md

```diff
@@ -3,122 +3,120 @@
+#=
+	Multi-line comments can be used for longer explanations.
 
-    This is most commonly used for one-line function definitions or mathematical functions, where the function body is a single expression.
+	They are especially useful to comment out blocks of code during debugging.
+=#
+```
 
-### Invoking functions
+## Variables and assignment
 
-Invoking a function is done by specifying its name and passing arguments for each of the function's parameters:
+To create a variable, just assign a value to it:
 
-```julia
-# invoking a function
-muladd(10, 5, 1)
+```julia-repl
+julia> myvar = 42  # an integer
+42
 
-# and of course you can invoke a function within the body of another function:
-square_plus_one(x) = muladd(x, x, 1)
+julia> name = "Maria"  # strings are surrounded by double-quotes ""
+"Maria"
 ```
 
-## Integers and Arithmetic operations
+Types are an important subject in Julia, but for now it is best to ignore them them.
+The compiler will infer a suitable type for any expression you use.
 
-Integer literals in Julia are represented in the usual way.
-Underscores may be used as a digit separator.
+## Constants
 
-```julia-repl
-julia> 1
-1
-
-julia> -1234
--1234
+Global variables, created outside any function, are:
+- Allowed.
+- Sometimes necessary.
+- Usually discouraged (only within `*.jl` files; the REPL operates differently).
 
-julia> 1_234_567_890
-1234567890
-```
+If a value needs to be available throughout the program, but is not expected to change, use a constant instead.
 
-### Arithmetic operations
+Prefacing the assignemt with the `const` keyword allows the compiler to generate more efficient code.
 
-The standard prefix and infix operations are available: `+`, `-`, `*`, `%`.
+Accidentally trying to change the `const` value will give a warning:
 
 ```julia-repl
-julia> +16
-16
+julia> const answer = 42
+42
 
-julia> -16
--16
-
-julia> 16 + 6
-22
+julia> answer = 24
+WARNING: redefinition of constant Main.answer. This may fail, cause incorrect answers, or produce other errors.
+24
+```
 
-julia> 16 - 6
-10
+## Arithmetic operators
 
-julia> 16 * 6
-96
+These mostly work conventionally:
 
-julia> 16 % 6
-4
+```julia
+2 + 3  # 5 (addition)
+2 - 3  # -1 (subtraction)
+2 * 3  # 6 (mutlplication)
+8 / 2  # 4.0 (division with floating-point result)
+8 % 3  # 2 (remainder)
 ```
 
-#### Division
+## Functions
 
-Dividing two numbers with the `/` operator will result in a floating point value.
-To perform integer division
+There are two common ways to define a named function in Julia:
 
-- use the `div(x, y)` function, or
-- use the `÷` operator (Julia source code is unicode-aware)
+1. Using the `function` keyword
 
-```julia-repl
-julia> 16 / 6
-2.6666666666666665
+    ```julia
+    function muladd(x, y, z)
+        x * y + z
+    end
+    ```
 
-julia> div(16, 6)
-2
+    Indentation by 4 spaces is conventional for readability, but the compiler ignores this.
+    The `end` keyword is essential: more like Ruby than like Python.
```

Comment:
Referring to Ruby or Python might be confusing. A student would need to know the syntax from both languages to understand this. We can simply say `end` is part of the required syntax for defining the function.

Reply:
Again, probably better for `about.md` than `introduction.md`

Reply:
Removed the bit about other languages.

Bethany is keen to see cross language comparison, and Julia is unlikely to be any student's first programming language. However, this is for `about.md`, not an introduction to the very first exercise.